sympy/integrals/transforms.py

```diff
@@ -1826,19 +1760,84 @@ def _collapse_extra(self, extra):
         return plane, cond
 
     def _try_directly(self, **hints):
+        T = None
+        try_directly = not any(func.has(self.function_variable)
+                               for func in self.function.atoms(AppliedUndef))
+        if try_directly:
+            try:
+                T = self._compute_transform(self.function,
+                    self.function_variable, self.transform_variable, **hints)
+            except IntegralTransformError:
+                T = None
+
+        return self.function, T
+
+    def doit(self, **hints):
+        """
+        Try to evaluate the transform in closed form.
+
+        Explanation
+        ===========
+
+        Standard hints are the following:
+        - ``noconds``:  if True, do not return convergence conditions. This is
+        the default behaviour.
+        """
+        _noconds = hints.get('noconds', True)
+
+        debug('[LT doit] (%s, %s, %s)'%(self.function,
+                                        self.function_variable,
+                                        self.transform_variable))
+
         fn = self.function
-        debug('----> _try_directly: %s'%(fn, ))
         t_ = self.function_variable
         s_ = self.transform_variable
-        LT = None
-        if not fn.is_Add:
-            fn = expand_mul(fn)
-            try:
-                LT = self._compute_transform(fn, t_, s_, **hints)
-            except IntegralTransformError:
-                LT = None
-        return fn, LT
 
+        if fn.has(Heaviside(t_)) and not fn.has(DiracDelta(t_)):
+            fn = fn.replace(Heaviside(t_), 1)
```

Comment:
Is this not handled by the Heaviside rule?

Reply:
Yes, it is. Change (C) in my list below.